crates/fluvio-version-manager/src/command/install.rs

```diff
@@ -0,0 +1,285 @@
+                    .await
+                    .map_err(|err| Error::msg(err.to_string()))?;
+
+                if shasum == upstream_shasum {
+                    Self::set_executable_mode(&mut binary)?;
+                } else {
+                    return Err(Error::msg(format!(
+                        "Artifact {} didnt matched upstream shasum {}. Skipping installation for this artifact...",
+                        artf.name, shasum
+                    )));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Allocates artifacts in the FVM `pkgset` directory for future use.
+    /// Returns the path to the allocated version directory.
+    async fn allocate_artifacts(&self, tmp_dir: &TempDir, pkgset: &PackageSet) -> Result<PathBuf> {
+        let version_path = fvm_versions_path()?.join(&self.version.to_string());
+
+        create_dir(&version_path)?;
+
+        for artif in pkgset.artifacts.iter() {
+            copy(
+                tmp_dir.path().join(&artif.name),
+                version_path.join(&artif.name),
+            )?;
+        }
+
+        Ok(version_path)
+    }
+
+    /// Writes out the manifest for the specified `PackageSet`
+    async fn write_out_manifest(
+        &self,
+        version_path: PathBuf,
+        pkgset: &PackageSet,
+        channel: &Channel,
+    ) -> Result<()> {
+        let manifest_path = version_path.join(PACKAGE_SET_MANIFEST_FILENAME);
+        let manifest = VersionManifest {
+            channel: channel.to_owned(),
+            version: pkgset.version.clone(),
+        };
+        let contents = serde_json::to_string(&manifest)?;
+
+        write(manifest_path, contents)?;
+
+        Ok(())
+    }
+
+    /// Generates the Sha256 checksum for the specified file
+    fn shasum256(file: &File) -> Result<String> {
+        let meta = file.metadata()?;
+        let mut file = file.clone();
+        let mut hasher = Sha256::new();
+        let mut buffer = vec![0; meta.len() as usize];
+
+        file.read_exact(&mut buffer)?;
+        hasher.update(buffer);
+
+        let output = hasher.finalize();
+        Ok(hex::encode(output))
+    }
+
+    /// Sets the executable mode for the specified file in Unix systems.
+    /// This is no-op in non-Unix systems.
+    fn set_executable_mode(file: &mut File) -> Result<()> {
+        if cfg!(unix) {
+            use std::os::unix::fs::PermissionsExt;
+
+            const EXECUTABLE_MODE: u32 = 0o700;
+
+            // Add u+rwx mode to the existing file permissions, leaving others unchanged
+            let mut permissions = file.metadata()?.permissions();
+            let mut mode = permissions.mode();
+
+            mode |= EXECUTABLE_MODE;
+            permissions.set_mode(mode);
+            file.set_permissions(permissions)?;
+
+            return Ok(());
+        }
+
+        Ok(())
+    }
+}
+
+impl Notify for InstallOpt {
+    fn is_quiet(&self) -> bool {
+        self.global_opts.quiet
+    }
+}
```

Comment:
Need to have unit tests for checksum, executable, and manifest logic.

Reply:
Checksum is now in the previous PR #3589. Im just missing Executable and Manifest!